File: Learning/CodeWars/Ruby/6kyu_WeIrD StRiNg CaSe.rb
# See: https://www.codewars.com/kata/52b757663a95b11b3d00062d

def weirdcase string
  string.split(' ').map do |word|
    word.split('').each_with_index.map do |item, index|
      index.even? ? item.upcase : item.downcase
    end.join('')
  end.join(' ')
end

File: Learning/CodeWars/JavaScript/6kyu_WeIrD_StRiNg_CaSe.js
// See: https://www.codewars.com/kata/52b757663a95b11b3d00062d

const toWeirdCase = function (string){
  return string.toLowerCase().split(' ').map(mapWord).join(' ');
};

const mapWord = function (word) {
  return word.split('').map(mapCharacter).join('');
};

const mapCharacter = function (letter, index) {
  return (index%2 === 0) ? letter.toUpperCase() : letter;
};